controllers/sensor/validate.go

```diff
@@ -469,32 +468,32 @@ func validateEventFilter(filter *v1alpha1.EventDependencyFilter) error {
 
 // validateEventTimeFilter validates time filter
 func validateEventTimeFilter(tFilter *v1alpha1.TimeFilter) error {
-	currentT := time.Now().UTC()
-	currentT = time.Date(currentT.Year(), currentT.Month(), currentT.Day(), 0, 0, 0, 0, time.UTC)
-	currentTStr := currentT.Format(common.StandardYYYYMMDDFormat)
+	now := time.Now().UTC()
 	if tFilter.Start != "" && tFilter.Stop != "" {
-		startTime, err := time.Parse(common.StandardTimeFormat, fmt.Sprintf("%s %s", currentTStr, tFilter.Start))
+		startTime, err := common.ParseTime(tFilter.Start, now)
 		if err != nil {
 			return err
 		}
-		stopTime, err := time.Parse(common.StandardTimeFormat, fmt.Sprintf("%s %s", currentTStr, tFilter.Stop))
+		stopTime, err := common.ParseTime(tFilter.Stop, now)
 		if err != nil {
 			return err
 		}
-		if stopTime.Before(startTime) || startTime.Equal(stopTime) {
-			return errors.Errorf("invalid event time filter: stop '%s' is before or equal to start '%s", tFilter.Stop, tFilter.Start)
+		if stopTime.Equal(startTime) {
+			return errors.Errorf("invalid event time filter: stop '%s' is equal to start '%s", tFilter.Stop, tFilter.Start)
 		}
+		return nil
+	}
+
+	if tFilter.Start != "" {
```

Comment:
If both `Start` and `Stop` == `""`, they will be parsed twice, should be able to only do once.

Also need to check if both of them `==""`.

Reply:
No, the entire structure of this function is like

```go
if Start != "" && Stop != "" {
	// parse Start and Stop
	return ...
}
if Start != "" { // it means `Stop == ""`
	// parse only Start
	return ...
}
if Stop != "" { // it means `Start ==""`
	// parse only Stop
	return ...
}
return nil
```

If this looks less straightforward, maybe this'd be better?

```go
if Start != "" && Stop != "" {
	// parse Start and Stop
	return ...
} else if Start != "" { // it means `Stop == ""`
	// parse only Start
	return ...
} else if Stop != "" { // it means `Start ==""`
	// parse only Stop
	return ...
} else {
	return nil
}
```